### PR TITLE
Add support for pluggable PM policies

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
arch: Kconfig: Add Kconfig options for Low Power States
soc: arm: nrf52: Select the supported Power States
subsys: power: Add support for pluggable PM policies
samples: subsys: power_mgr: Add changes as per PM policy Kconfig

### CODE REVIEW

File: subsys/power/Kconfig [hunk 1]
```diff
@@ -8,39 +8,16 @@ config PM_CONTROL_OS_LPS
...
-	int "LPS minimum residency"
-	default 5
-	help
-	  Minimum residency in ticks to enter LPS state.
-endif
-
 config PM_CONTROL_OS_LPS_1
 	bool "Platform supports LPS_1"
 	help
 	  Select this option if SoC support LPS_1 state.
```
Comment (pizi-nordic) [inline subsys/power/Kconfig]: Side comment: 

For me the naming of states creates some confusion. I really do now know how many states are in fact supported on my platform and how Zephyr names maps to platform capabilities. 

IMHO each platform which supports any kind of power management should define own list of supported states (Example: my platform support LPS_1, LPS_2 and DEEP_SLEEP) and provide some information how they map to the names in datasheet. User should be able to disable some states in Kconfig, however we should not force him to solve multi-selection test about power management.

File: subsys/power/policy/Kconfig [hunk 2]
```diff
@@ -0,0 +1,69 @@
+
+choice
+	prompt "PM Policy"
```
Comment (pizi-nordic) [inline subsys/power/policy/Kconfig]: I assume that this is in fact "Idle State management".
Could we name things accordingly in order to be prepared for other features, like Frequency/Voltage scaling etc.?

For example:
- Power Management Options:
  - Idle State Management
     - Idle state handling policy
  - CPU Frequency scaling [example]
  - ....

File: subsys/power/Kconfig (hunk 1 above)
Comment (ramakrishnapallala) [reply subsys/power/Kconfig]: @pizi-nordic  I think we can add the individual supported states as Kconfig option at arch level like SYS_POWER_LOW_POWER_STATE_SUPPORTED and SoC implementations can select them based on the SoC specific support.

So, in policy Kconfig we make them depend on the arch level Kconfig, that way we will only have what is supported by a SoC.

Comment (pizi-nordic) [reply subsys/power/Kconfig]: I think that it is a very good idea.

**Comment by ramakrishnapallala** [comment]
@pizi-nordic I have updated the PR with proposed changes. Please review the changes.

**Review by pizi-nordic — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**